src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/FunctionPointerCallingConventionClassifier.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Classification.Classifiers;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Classification
+{
+    internal class FunctionPointerCallingConventionClassifier : AbstractSyntaxClassifier
+    {
+        public override ImmutableArray<Type> SyntaxNodeTypes { get; } = ImmutableArray.Create(
+            typeof(FunctionPointerUnmanagedCallingConventionSyntax));
+
+        public override void AddClassifications(
+            SyntaxNode syntax,
+            SemanticModel semanticModel,
+            ClassificationOptions options,
+            ArrayBuilder<ClassifiedSpan> result,
+            CancellationToken cancellationToken)
+        {
+            // We may use semanticModel.GetSymbolInfo if https://github.com/dotnet/roslyn/issues/59060 got fixed.
+            var name = ((FunctionPointerUnmanagedCallingConventionSyntax)syntax).Name;
+            var type = semanticModel.Compilation.GetTypeByMetadataName("System.Runtime.CompilerServices.CallConv" + name.ValueText);
```

Comment:
_If_ you're going to do this (and honestly I would not advise it, as it is incorrect for the single convention case), this is not the correct API. The compiler only considers types that come from the corlib assembly, not from any assembly.

Separately, for the single-convention case where it's just `Cdecl`, `Stdcall`, `Thiscall`, or `Fastcall`, this is incorrect. The compiler does not look for types at all in these cases, and they might not even exist on some platforms.

Reply:
@333fred Do you mean the new IDE behavior should be done for multi-convention only?

Reply:
> @333fred Do you mean the new IDE behavior should be done for multi-convention only?

and also those that are not special-cased (anything other than Cdecl, Stdcall, Thiscall, Fastcall)